File: EncontraMaiorNumero.java
public class EncontraMaiorNumero {
    public static void main(String[] args) {
        // criando a array com os números
        int[] lista = { 1, 3, 5, 7, 2, 8, 4, 9, 6, 10 };

        // iniciando a variavel pra guardar o numero maior
        int maior = lista[0];

        // percorrendo a lista pra encontrar o numero maior
        for (int i = 1; i < lista.length; i++) {
            if (lista[i] > maior) {
                maior = lista[i]
            }
        }
        System.out.println("O maior número é:" + maior);
    }
}

/*
 * Sintaxe do loop for em java:
 * 
 * for (inicialização; condição; incremento) {
 * código a ser executado
 * }
 * 
 * 1. Inicialização:
 * 
 * A parte de inicialização é executada só uma vez, antes do início do loop
 * No código a inicialização é: int i = 1;
 * Isso cria uma variável i e inicia ela com o valor 1
 * 
 * 2. Condição:
 * 
 * A condição é verificada antes de cada iteração do loop
 * Se a condição for verdadeira, o bloco de código dentro do loop é executado;
 * caso contrário, o loop é encerrado.
 * No código, a condição é: i < lista.length;
 * Isso verifica se o valor de i é menor que o tamanho da lista
 * 
 * 3. Incremento:
 * 
 * A parte de incremento é executada após cada iteração do loop
 * No código, o incremento é: i++
 * Isso aumenta o valor de i em 1 a cada iteração
 * 
 * 4. Execução do "código a ser executado":
 * 
 * O código dentro do loop é executado enquanto a condição for verdadeira.
 * No código fornecido, o código a ser executado é:
 * 
 * if (lista[i] > maior) {
 * maior = lista[i];
 * }
 * 
 * Isso verifica se o número atual (na posição i da lista) é maior que o valor
 * atual de "maior"
 * Se for, atualiza o valor de maior para o número atual
 */

File: EncontraMaiorNumero/EncontraMaiorNumero.py
# Criando a lista com os números
lista = [1, 3, 5, 7, 2, 8, 4, 9, 6, 10]

# Iniciando a variável para guardar o número maior
maior = lista[0]

# Percorrendo a lista para encontrar o número maior
for num in lista[1:]:
    if num > maior:
        maior = num

print(f"O maior número é: {maior}")

''' Sintaxe do Loop for em Python
Em Python, o loop for é usado para iterar sobre uma sequência (como uma lista, uma string ou um intervalo numérico)
A sintaxe básica é a seguinte:

for elemento in sequencia:
    # Código a ser executado

Inicialização (ausente em Python):
Diferentemente do Java, em Python, não precisamos especificar uma inicialização separada. O loop for automaticamente itera sobre os elementos da sequência.

Condição (ausente em Python):
Também não especificamos explicitamente uma condição. O for percorre todos os elementos da sequência até que não haja mais elementos.

Incremento (ausente em Python):
Não precisamos incrementar manualmente o índice ou contador. O for cuida disso automaticamente.

Execução do “código a ser executado”:
O bloco de código dentro do for é executado para cada elemento da sequência.
O elemento representa cada item da sequência.

Exemplo em Python
Suponha que temos a seguinte lista:
numeros = [1, 2, 3, 4, 5]

Podemos usar o for para percorrer essa lista:
for num in numeros:
    print(num)

Nesse exemplo, o código dentro do for será executado 5 vezes (uma vez para cada número na lista), e cada número será impresso no console.
O Python simplifica a sintaxe do for, tornando-o mais legível e conciso.
'''